src/gates/curve_add.rs

```diff
@@ -36,37 +36,50 @@ impl<C: HaloCurve, InnerC: Curve<BaseField = C::ScalarField>> CurveAddGate<C, In
 impl<C: HaloCurve, InnerC: Curve<BaseField = C::ScalarField>> Gate<C> for CurveAddGate<C, InnerC> {
     const NAME: &'static str = "CurveAddGate";
 
-    const PREFIX: &'static [bool] = &[true, false, true, false, false, false];
+    const PREFIX: &'static [bool] = &[true, false, true, false, true];
 
     fn evaluate_unfiltered(
         _local_constant_values: &[InnerC::BaseField],
         local_wire_values: &[InnerC::BaseField],
         right_wire_values: &[InnerC::BaseField],
         _below_wire_values: &[InnerC::BaseField],
     ) -> Vec<InnerC::BaseField> {
+        // Notation:
+        // - p1 is the accumulator;
+        // - p2 is the addend;
+        // - p3 = p1 + p2;
+        // - p4 = if scalar_bit { p4 } else { p1 }
+
         let x1 = local_wire_values[Self::WIRE_GROUP_ACC_X];
         let y1 = local_wire_values[Self::WIRE_GROUP_ACC_Y];
-        let x3 = right_wire_values[Self::WIRE_GROUP_ACC_X];
-        let y3 = right_wire_values[Self::WIRE_GROUP_ACC_Y];
+        let x4 = right_wire_values[Self::WIRE_GROUP_ACC_X];
+        let y4 = right_wire_values[Self::WIRE_GROUP_ACC_Y];
 
         let scalar_acc_old = local_wire_values[Self::WIRE_SCALAR_ACC_OLD];
         let scalar_acc_new = local_wire_values[Self::WIRE_SCALAR_ACC_NEW];
         let x2 = local_wire_values[Self::WIRE_ADDEND_X];
         let y2 = local_wire_values[Self::WIRE_ADDEND_Y];
-        let scalar_bit = C::ScalarField::ONE;
+        let scalar_bit = local_wire_values[Self::WIRE_SCALAR_BIT];
         let inverse = local_wire_values[Self::WIRE_INVERSE];
         let lambda = local_wire_values[Self::WIRE_LAMBDA];
 
         let computed_lambda = (y1 - y2) * inverse;
-        let computed_x3 = lambda.square() - x1 - x2;
-        let computed_y3 = lambda * (x1 - x3) - y1;
+        let x3 = lambda.square() - x1 - x2;
+        // We subtract x4 instead of x3 in order to minimize degree. This will give an incorrect
+        // result for y3 if x3 != x4, which happens when scalar_bit = 0, but in that case y3 will
+        // be ignored (i.e. multiplied by zero), so we're okay.
+        let y3 = lambda * (x1 - x4) - y1;
+
+        let not_scalar_bit = scalar_bit - InnerC::BaseField::ONE;
```

Comment:
Should be `not_scalar_bit = InnerC::BaseField::ONE - scalar_bit` I think.

Reply:
Good catch!